sky/authentication.py

```diff
@@ -267,10 +268,33 @@ def setup_gcp_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
     return _replace_ssh_info_in_config(config, public_key)
 
 
+# In Azure, cloud-init script must be encoded in base64. See
+# https://learn.microsoft.com/en-us/azure/virtual-machines/custom-data
+# for more information. Here we decode it and replace the ssh user
+# and public key content, then encode it back.
+def _replace_ssh_info_in_cloud_init(config: Dict[str, Any],
```

Comment:
This seems only be used by Azure, should we have it inline in `setup_azure_authentication`?